packages/datadog-plugin-aerospike/test/index.spec.js

```diff
@@ -0,0 +1,300 @@
+'use strict'
+
+const agent = require('../../dd-trace/test/plugins/agent')
+const { ERROR_MESSAGE, ERROR_TYPE, ERROR_STACK } = require('../../dd-trace/src/constants')
+
+describe('Plugin', () => {
+  let client
+  let aerospike
+  let config
+  let tracer
+  let ns
+  let set
+  let userKey
+  let key
+  let keyString
+
+  describe('aerospike', () => {
+    withVersions('aerospike', 'aerospike', version => {
+      beforeEach(() => {
+        tracer = require('../../dd-trace')
+        aerospike = require(`../../../versions/aerospike@${version}`).get()
+      })
+
+      beforeEach(() => {
+        ns = 'test'
+        set = 'demo'
+        userKey = 'key'
+
+        config = {
+          hosts: '127.0.0.1:3000',
+          port: '3000'
+        }
+
+        client = aerospike.client(config)
+        key = new aerospike.Key(ns, set, userKey)
+        keyString = `${ns}:${set}:${userKey}`
+      })
+
+      after(() => {
+        return agent.close({ ritmReset: false })
+      })
+
+      describe('without configuration', () => {
+        before(() => {
+          return agent.load('aerospike')
+        })
+
+        afterEach(() => {
+          client.close()
+        })
+
+        describe('client', () => {
+          it('should instrument put', done => {
+            agent
+              .use(traces => {
+                const span = traces[0][0]
+                const resource = `Put`
+                expect(span).to.have.property('name', 'aerospike.command')
+                expect(span).to.have.property('service', 'test-aerospike')
+                expect(span).to.have.property('resource', resource)
+                expect(span).to.have.property('type', 'aerospike')
+                expect(span.meta).to.have.property('span.kind', 'client')
+                expect(span.meta).to.have.property('aerospike.key', keyString)
+                expect(span.meta).to.have.property('aerospike.namespace', ns)
+                expect(span.meta).to.have.property('aerospike.setname', set)
+                expect(span.meta).to.have.property('aerospike.userkey', userKey)
+                expect(span.meta).to.have.property('component', 'aerospike')
+              })
+              .then(done)
+              .catch(done)
+
+            client.connect(() => {
+              client.put(key, { i: 123 }, () => {})
+            })
+          })
+
+          it('should instrument connect', done => {
```

Comment:
This one is interesting because it might happen outside of a request. Is it explicitly sent by the user or is it an implicit command sent automatically when other types of commands are sent?

Reply:
Well I believe it is always sent because aerospike works by chaining callbacks and promises, so you would first always have to connect and then on the promise/callback of that call you can perform other requests inside. But no I don't think you connect first and then later on run a request, everything has to be chained in the connects call back